src/cpp/include/openvino/genai/vision_encoder.hpp

```diff
@@ -41,11 +41,16 @@ struct EncodedImage {
 /// ov::InferRequest and configured by ProcessorConfig.
 class OPENVINO_GENAI_EXPORTS VisionEncoder {
 public:
+    /// @brief A string denoting model type.
+    std::string model_type;
```

Comment:
Can this be a `class enum`? That way we additionally describe the list of supported models.

Reply:
Added enum class for model type